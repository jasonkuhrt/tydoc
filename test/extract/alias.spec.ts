it('raw is based on type alias declaration node', () => {
  expect(
    ctx.extract(`
      export { A }
      /**
       * foobar
       */
      type A = {}
  `)
  ).toMatchInlineSnapshot(`
    Object {
      "modules": Array [
        Object {
          "kind": "module",
          "location": Object {
            "absoluteFilePath": "/a.ts",
          },
          "mainExport": null,
          "name": "a",
          "namedExports": Array [
            Object {
              "isTerm": false,
              "isType": true,
              "kind": "export",
              "name": "A",
              "type": Object {
                "kind": "typeIndexRef",
                "link": "(\\"/a\\").A",
              },
            },
          ],
        },
      ],
      "typeIndex": Object {
        "(\\"/a\\").A": Object {
          "kind": "alias",
          "name": "A",
          "raw": Object {
            "nodeFullText": "/**
     * foobar
     */
    type A = {};",
            "nodeText": "type A = {};",
            "typeText": "A",
          },
          "type": Object {
            "kind": "object",
            "props": Array [],
            "raw": Object {
              "nodeFullText": "/**
     * foobar
     */
    type A = {};",
              "nodeText": "type A = {};",
              "typeText": "A",
            },
          },
        },
      },
    }
  `)
})

it('type alias of number is added to type index', () => {
  expect(
    ctx.extract(`
      export type A = 1
    `)
  ).toMatchInlineSnapshot(`
    Object {
      "modules": Array [
        Object {
          "kind": "module",
          "location": Object {
            "absoluteFilePath": "/a.ts",
          },
          "mainExport": null,
          "name": "a",
          "namedExports": Array [
            Object {
              "isTerm": false,
              "isType": true,
              "kind": "export",
              "name": "A",
              "type": Object {
                "kind": "typeIndexRef",
                "link": "(\\"/a\\").A",
              },
            },
          ],
        },
      ],
      "typeIndex": Object {
        "(\\"/a\\").A": Object {
          "kind": "alias",
          "name": "A",
          "raw": Object {
            "nodeFullText": "",
            "nodeText": "",
            "typeText": "1",
          },
          "type": Object {
            "base": "number",
            "kind": "literal",
            "name": "1",
          },
        },
      },
    }
  `)
})
