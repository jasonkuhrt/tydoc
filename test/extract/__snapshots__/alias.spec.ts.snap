// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exported type alias of function is added to type index 1`] = `
Object {
  "modules": Array [
    Object {
      "kind": "module",
      "location": Object {
        "absoluteFilePath": "/a.ts",
      },
      "mainExport": null,
      "name": "a",
      "namedExports": Array [
        Object {
          "isTerm": false,
          "isType": true,
          "kind": "export",
          "name": "A",
          "type": Object {
            "kind": "typeIndexRef",
            "link": "(\\"/a\\").A",
          },
        },
      ],
    },
  ],
  "typeIndex": Object {
    "(\\"/a\\").A": Object {
      "kind": "alias",
      "name": "A",
      "raw": Object {
        "nodeFullText": "export type A = () => {};",
        "nodeText": "export type A = () => {};",
        "typeText": "A",
      },
      "type": Object {
        "hasProps": false,
        "isOverloaded": false,
        "kind": "callable",
        "props": Array [],
        "raw": Object {
          "nodeFullText": "export type A = () => {};",
          "nodeText": "export type A = () => {};",
          "typeText": "A",
        },
        "sigs": Array [
          Object {
            "kind": "sig",
            "params": Array [],
            "return": Object {
              "kind": "object",
              "props": Array [],
              "raw": Object {
                "nodeFullText": "",
                "nodeText": "",
                "typeText": "{}",
              },
            },
          },
        ],
      },
    },
  },
}
`;

exports[`exported type alias of function via typeof is added to type index 1`] = `
Object {
  "modules": Array [
    Object {
      "kind": "module",
      "location": Object {
        "absoluteFilePath": "/a.ts",
      },
      "mainExport": null,
      "name": "a",
      "namedExports": Array [
        Object {
          "isTerm": false,
          "isType": true,
          "kind": "export",
          "name": "A",
          "type": Object {
            "kind": "typeIndexRef",
            "link": "(\\"/a\\").A",
          },
        },
      ],
    },
  ],
  "typeIndex": Object {
    "(\\"/a\\").A": Object {
      "kind": "alias",
      "name": "A",
      "raw": Object {
        "nodeFullText": "
export type A = typeof a;",
        "nodeText": "export type A = typeof a;",
        "typeText": "() => void",
      },
      "type": Object {
        "hasProps": false,
        "isOverloaded": false,
        "kind": "callable",
        "props": Array [],
        "raw": Object {
          "nodeFullText": "() => {}",
          "nodeText": "() => {}",
          "typeText": "() => void",
        },
        "sigs": Array [
          Object {
            "kind": "sig",
            "params": Array [],
            "return": Object {
              "kind": "primitive",
              "type": "void",
            },
          },
        ],
      },
    },
  },
}
`;

exports[`exported type alias of object is added to type index 1`] = `
Object {
  "modules": Array [
    Object {
      "kind": "module",
      "location": Object {
        "absoluteFilePath": "/a.ts",
      },
      "mainExport": null,
      "name": "a",
      "namedExports": Array [
        Object {
          "isTerm": false,
          "isType": true,
          "kind": "export",
          "name": "A",
          "type": Object {
            "kind": "typeIndexRef",
            "link": "(\\"/a\\").A",
          },
        },
      ],
    },
  ],
  "typeIndex": Object {
    "(\\"/a\\").A": Object {
      "kind": "alias",
      "name": "A",
      "raw": Object {
        "nodeFullText": "export type A = { b: 1 };",
        "nodeText": "export type A = { b: 1 };",
        "typeText": "A",
      },
      "type": Object {
        "kind": "object",
        "props": Array [
          Object {
            "kind": "prop",
            "name": "b",
            "type": Object {
              "base": "number",
              "kind": "literal",
              "name": "1",
            },
          },
        ],
        "raw": Object {
          "nodeFullText": "export type A = { b: 1 };",
          "nodeText": "export type A = { b: 1 };",
          "typeText": "A",
        },
      },
    },
  },
}
`;

exports[`exported type alias of object via typeof is added to type index 1`] = `
Object {
  "modules": Array [
    Object {
      "kind": "module",
      "location": Object {
        "absoluteFilePath": "/a.ts",
      },
      "mainExport": null,
      "name": "a",
      "namedExports": Array [
        Object {
          "isTerm": false,
          "isType": true,
          "kind": "export",
          "name": "A",
          "type": Object {
            "kind": "typeIndexRef",
            "link": "(\\"/a\\").A",
          },
        },
      ],
    },
  ],
  "typeIndex": Object {
    "(\\"/a\\").A": Object {
      "kind": "alias",
      "name": "A",
      "raw": Object {
        "nodeFullText": "
export type A = typeof a;",
        "nodeText": "export type A = typeof a;",
        "typeText": "{ b: number; }",
      },
      "type": Object {
        "kind": "object",
        "props": Array [
          Object {
            "kind": "prop",
            "name": "b",
            "type": Object {
              "kind": "primitive",
              "type": "number",
            },
          },
        ],
        "raw": Object {
          "nodeFullText": "{ b: 1 }",
          "nodeText": "{ b: 1 }",
          "typeText": "{ b: number; }",
        },
      },
    },
  },
}
`;
